### DISCUSSION TITLE
Universal provider with the new web3modal

### DISCUSSION THREAD

**Discussion by kieranbop1990** [WalletConnect/walletconnect-monorepo#3643] [Developer Support] [answered]
Are there any examples or plans for the new web3modal to work with the universal provider? I had to move to the universal provider when going from v1 -> v2 and I lost the ability to use metamask browser integration for example.

**Comment by glitch-txs** [comment]
hi, why did you move to Unviersal instead of EthereumProvider? Are you using non EVM chains? or was it because of lack of ethers/web3.js support?

**Reply by kieranbop1990** [reply]
Yeah I am using both.

**Reply by glitch-txs** [reply]
We'll add ethers support for Web3Modal v3 soon

**Reply by kieranbop1990** [reply]
I followed the ethers and v2 example here https://github.com/WalletConnect/web-examples/tree/main/dapps a fair bit. So I could possibly swap out the web3modal with the universal provider soon and get some more desktop functionality back? Thanks

**Reply by kieranbop1990** [reply]
Or potentially use a hybrid solution depending on what is requested use the ethereumprovider else universal?

**Reply by glitch-txs** [reply]
You'll be able to use the full Web3Modal v3 SDK once we add support with ethers, so you won't need Universal Provider

**Reply by kieranbop1990** [reply]
Fantastic. Can I ask a rough timeline so I can check back or something I can track regarding progress? I am happy to contribute some work towards it?

**Reply by glitch-txs** [reply, marked as answer]
We'll start working on this shortly, you can check the branch V3 [here](https://github.com/WalletConnect/web3modal/tree/V3). There's no ETA atm

**Reply by kieranbop1990** [reply]
is v3 only using wagmi or will there be ones without it?

**Reply by glitch-txs** [reply]
there's going to be a package for wagmi and other one for ethers (without wagmi). Feel free to ping @0xAsimetriq for questions related to OSS collab :)
